ls1intum/Artemis
# Feature/markdown editor/katex support

### Checklist

### Motivation and Context



Some lectures / exercises want to mathematical formulas in the problem statements, etc.
The idea is to include the well known [katex](https://katex.org/) lib to use this kind of typesets.

### Description


This PR adds a command to the markdown editor to insert a katex formula template (see screenshot below).
- The command is a domain command, because components that provide their own previews might not be able to parse it.
- The parsing is done using `showdown-katex`
- The implementation for `remarkable-katex` did not work properly for me (weird styling issues), therefore **katex is not available for programming exercises atm** (migrating the remarkable parser to showdown is out of scope for this PR)
- I had to use a different DomSanitizer (DOMPurify), as the Angular Sanitizer destroyed the katex styles.

:warning: This PR affects the preview rendering in all files that use the artemisMarkdown service for rendering markdown into html. :warning: 

### Steps for Testing


1. Log in to ArTEMiS
2. Open Create/Edit Text-Exercise / Modeling-Exercise Dialog
3. Insert formula and check preview

### Screenshots



![image](https://user-images.githubusercontent.com/28230611/60972462-a0995980-a326-11e9-8f68-6d61f12dc245.png)

![image](https://user-images.githubusercontent.com/28230611/60972487-ae4edf00-a326-11e9-82d3-d256cc989649.png)
